src/v/cloud_storage/cache_service.h

```diff
@@ -47,10 +51,30 @@ class cache {
     /// Remove element from cache by key
     ss::future<> invalidate(const std::filesystem::path& key);
 
+    // Total cleaned is exposed for better testability of eviction
+    uint64_t get_total_cleaned();
+
 private:
+    /// Triggers directory walker, creates a list of files to delete and deletes
+    /// them until cache size <= _cache_size_low_watermark * max_cache_size
+    ss::future<> clean_up_cache();
+
+    /// Triggers directory walker, creates a list of files to delete and deletes
+    /// only tmp files that are left from previous Red Panda run
+    ss::future<> clean_up_at_start();
+
+    bool is_tmp_file(const ss::sstring& filename);
```

Comment:
this can be a static function inside the `.cc` file, no need to make it an instance method